cli/client.go

```diff
@@ -0,0 +1,247 @@
+package main
```

Comment:
This is very similar to the previous `client.go`, it's just in a different folder. The main difference here, though, is that we use Go objects that are shared between the client and the server for business logic rather than having two structs with the same name and duplicated logic in both codebases.